### PR TITLE
[pt2] add meta for `linalg_ldl_solve`

### PR DESCRIPTION
**Pull request by nkaretnikov**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #101375
* #101372
* __->__ #101367
* #101362
* #101315

### CODE REVIEW

**Comment by nkaretnikov** [comment]
Part of this stack because it unlocks tests next to the prev PR in the stack.

File: torch/_meta_registrations.py [hunk 1]
```diff
@@ -564,6 +564,46 @@ def linalg_ldl_factor_ex_meta(
...
+    )
+    B_broadcast_size, _ = _linalg_broadcast_batch_dims(B, LD)
+    return torch.empty_strided(
+        size=B_broadcast_size,
+        stride=make_contiguous_strides_for(B_broadcast_size, row_major=False),
+        dtype=B.dtype,
+        device=B.device,
+    )
+
+
```
Comment (nkaretnikov) [inline torch/_meta_registrations.py:606]: See `TORCH_META_FUNC(linalg_ldl_solve)` in core.

File: torch/_meta_registrations.py [hunk 2]
```diff
@@ -564,6 +564,46 @@ def linalg_ldl_factor_ex_meta(
...
+        ),
+    )
+    check(
+        utils.is_integer_dtype(pivots.dtype),
+        lambda: f"torch.linalg.ldl_solve: Expected pivots to be integers. Got {pivots.dtype}",
+    )
+    check(
+        LD.dtype == B.dtype,
+        lambda: f"torch.linalg.ldl_solve: LD dtype {LD.dtype} does not match b dtype {B.dtype}",
+    )
```
Comment (nkaretnikov) [inline torch/_meta_registrations.py:597]: `linearSolveCheckInputs` already has a dtype check

File: torch/_meta_registrations.py [hunk 3]
```diff
@@ -564,6 +564,46 @@ def linalg_ldl_factor_ex_meta(
...
+    squareCheckInputs(LD, "torch.linalg.ldl_solve")
+    checkFloatingOrComplex(LD, "torch.linalg.ldl_solve")
+    linearSolveCheckInputs(B, LD, "torch.linalg.ldl_solve")
+    check(
+        B.ndim >= 2,
+        lambda: (
+            f"torch.linalg.ldl_solve: Expected B to have at least 2 dimensions, "
+            f"but it has {B.ndim} dimensions instead"
+        ),
+    )
```
Comment (nkaretnikov) [inline torch/_meta_registrations.py:581]: This `ndim` check will never be reached because `self.size(-2)` is already checked in `linearSolveCheckInputs`

**Review by lezcano — APPROVED** [review]

**Pull request closed without merging**